plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1645,26 +1726,52 @@ private function set_order_prop( \WC_Abstract_Order $order, string $prop_name, $
 	}
 
 	/**
-	 * Return order data for a single order ID.
+	 * Retrieve raw order data for multiple IDs.
 	 *
-	 * @param int $id Order ID.
+	 * @param int[] $ids List of order IDs.
 	 *
-	 * @return object|\WP_Error DB order object or WP_Error.
+	 * @return \stdClass[] DB Order objects or error.
 	 */
-	private function get_order_data_for_id( $id ) {
-		$results = $this->get_order_data_for_ids( array( $id ) );
+	protected function get_order_data_for_ids( $ids ): array {
```

Comment:
I was hesitant to do this because `OrdersTableDataStore::get_orders_type()` doesn't strictly declare the argument type.  I made a change to that caller to temporarily handle this potential compatibility issue in hopes we can eventually push strict typing up the call stack.